packages/orbit-components/src/Stack/index.tsx

```diff
@@ -132,17 +139,21 @@ const Stack = (props: Props) => {
 
     return cx(
       typeof spaceAfter !== "undefined" && getSpaceAfterClasses(spaceAfter, viewport),
-      typeof align !== "undefined" && getAlignItemsClasses(align, viewport),
-      typeof align !== "undefined" && getAlignContentClasses(align, viewport),
-      typeof wrap !== "undefined" && getWrapClasses(wrap, viewport),
-      typeof grow !== "undefined" && getGrowClasses(grow, viewport),
-      typeof shrink !== "undefined" && getShrinkClasses(shrink, viewport),
-      typeof justify !== "undefined" && getJustifyClasses(justify, viewport),
-      getDirectionClasses(direction, viewport),
+      flex || inline
+        ? [
+            typeof align !== "undefined" && getAlignItemsClasses(align, viewport),
+            typeof align !== "undefined" && getAlignContentClasses(align, viewport),
+            typeof wrap !== "undefined" && getWrapClasses(wrap, viewport),
+            typeof grow !== "undefined" && getGrowClasses(grow, viewport),
+            typeof shrink !== "undefined" && getShrinkClasses(shrink, viewport),
+            typeof justify !== "undefined" && getJustifyClasses(justify, viewport),
+            getDirectionClasses(direction, viewport),
+          ]
+        : "block",
       getSpacingClasses(spacing, viewport, direction, legacy),
-      inline && getDisplayInlineClass(inline, viewport),
+      inline && getDisplayClasses("inline-flex", viewport),
+      flex && getDisplayClasses("flex", viewport),
```

Comment:
This might cause a conflict if both `inline` and `flex` props are true, as it will apply both classes. I had thought about that and that is why I suggested that the function received the props as arguments. Why did you decide to do it differently?

Also, docs state, about the `inline` prop:
> If `true`, the Stack will have `display` set to `inline-flex`, otherwise it will be `flex`. 

But this is incorrect. I would remove everything after the comma.

Reply:
that does not make sense to apply them both

Reply:
It might not but:
- Nothing prevents them from doing it (and we're not adding conditional types)
- There was no problem in the past because SC would resolve the conflict, but now nothing resolves the conflicts, so we must do it ourselves